core/config/engine.cpp

```diff
@@ -191,11 +191,11 @@ String Engine::get_architecture_name() const {
 #elif defined(__i386) || defined(__i386__) || defined(_M_IX86)
 	return "x86_32";
 
-#elif defined(__aarch64__) || defined(_M_ARM64)
+#elif defined(__aarch64__) || defined(_M_ARM64) || defined(_M_ARM64EC)
 	return "arm64";
 
-#elif defined(__ARM_ARCH_7A__) || defined(__ARM_ARCH_7S__)
-	return "armv7";
+#elif defined(__ARM_ARCH_7A__) || defined(__ARM_ARCH_7S__) || defined(_M_ARM)
+	return "arm32";
```

Comment:
This is likely also needed in `core/os/os.cpp`'s `OS::has_feature()`.

And the missed arch change is another confirmation that these two methods should really be unified as I advocated for #63258.

Reply:
`OS::has_feature()` already handle `arm32` feature and have `_M_ARM` define.

Reply:
I was confused because it uses seemingly different logic: https://github.com/godotengine/godot/blob/master/core/os/os.cpp#L441

This one here doesn't seem to handle `__arm__`?

Reply:
> This one here doesn't seem to handle __arm__?

I'm not sure which compiler/platform is using this define, Windows seems to be using `_M_ARM`.

Reply:
It's defined by GCC/Clang and other `__GNUC__` compatible compilers. But it's not directly related to this PR, just a reminder for myself / anyone interested to handle when unifying `OS::has_feature` and `Engine::get_architecture_name`.